2013-07-01	Nicolas Setton  <setton@adacore.com>

	* cairo-xlib-source: Use the core engine to render gradiants
	and fill rectangles, since XRenderCreateSolidFill and
	XRenderFillRectangles do not exist on older versions of UNIX

	(patch to be removed when upgrading to newer base version of
	platforms)

--- src/cairo-xlib-source.c.orig	2013-07-01 16:35:38.000000000 +0200
+++ src/cairo-xlib-source.c	2013-07-01 16:39:02.000000000 +0200
@@ -456,9 +456,9 @@
     xcolor.blue  = color->blue_short;
     xcolor.alpha = color->alpha_short;
 
-    if (CAIRO_RENDER_HAS_GRADIENTS(dst->display)) {
+    /* if (CAIRO_RENDER_HAS_GRADIENTS(dst->display)) {
 	picture = XRenderCreateSolidFill (dpy, &xcolor);
-    } else {
+    } else */ {
 	XRenderPictureAttributes pa;
 	int mask = 0;
 
@@ -470,10 +470,10 @@
 					_cairo_xlib_display_get_xrender_format (dst->display, CAIRO_FORMAT_ARGB32),
 					mask, &pa);
 
-	if (CAIRO_RENDER_HAS_FILL_RECTANGLES(dst->display)) {
+	/* if (CAIRO_RENDER_HAS_FILL_RECTANGLES(dst->display)) {
 	    XRectangle r = { 0, 0, 1, 1};
 	    XRenderFillRectangles (dpy, PictOpSrc, picture, &xcolor, &r, 1);
-	} else {
+	} else */ {
 	    XGCValues gcv;
 	    GC gc;
 
